### ISSUE TITLE
Q: when is rsync triggered, and how to re-trigger 

### ISSUE THREAD

**Issue by pvalena** [teemtee/tmt#1143]
When running `tmt run`, I am sometimes debugging CI and want to re-run the plan, with modified data, but don't want to re-install everything in the VM (i'm fine with re-running prepare, as it just check for packages).

ex.
```

$ tmt run -avv --skip finish provision -h virtual
[fails -> fix]
$ tmt run -l -avv --skip finish discover -f execute -f prepare -f report -f
[ is rsync run? ]
```

Currently, I think even if it's run, it's not run with '--del' so my pipeline fails on re-run prepare, as there are build leftovers. But it would be definitely good to know (e.g. with some --help) how to re-run workdir `rsync` on demand.

**Comment by psss** [comment]
Currently `guest.push()` is called before tests execution, guest preparation and before starting the interactive shell and by default `rsync` is called with the `--delete` option. Let's have a simple example plan :

```yaml
/plan:
    discover:
        how: fmf
    provision:
        how: virtual
    execute:
        how: tmt

/test:
    test: ls foo || touch foo
```

Here's the first execution:

```
> tmt run --until report execute -vvv
/var/tmp/tmt/run-226

/plan
    discover
        how: fmf
        directory: /tmp/debug
        summary: 1 test selected
    provision
        how: virtual
        image: fedora
        user: root
        memory: 2048 MB
        disk: 10 GB
        progress: booting...
        distro: Fedora Linux 35 (Cloud Edition)
        summary: 1 guest provisioned
    prepare
        summary: 0 preparations applied
    execute
        how: tmt
        order: 50
        exit-first: False
            test: /test
                out: ls: cannot access 'foo': No such file or directory
                00:00:00 /test [1/1]
        summary: 1 test executed
    report
        how: display
        summary: 1 test passed

total: 1 test passed
```

Running the test again with both `discover` and `execute` in `--force` mode confirms that the created file is deleted:

```
> tmt run --last --until report discover -f execute -fvvv
/var/tmp/tmt/run-226

/plan
    discover
        how: fmf
        directory: /tmp/debug
        summary: 1 test selected
    provision
        status: done
        summary: 1 guest provisioned
    prepare
        status: done
        summary: 0 preparations applied
    execute
        how: tmt
        order: 50
        exit-first: False
            test: /test
                out: ls: cannot access 'foo': No such file or directory
                00:00:00 /test [1/1]
        summary: 1 test executed
    report
        how: display
        summary: 1 test passed

total: 1 test passed
```

I tried your exact command line and it works for me as well. Are you able to reproduce your problem with the above-mentioned example as well?

**Comment by psss** [comment]
Btw, using `tmt run -vvvddd` will show all actions happenning under the hood including the `rsync` commands used for pushing workdir to guest. So you in this way you can verify what exactly is going on.

**Comment by pvalena** [comment]
Thanks for the response... would be actually nice to have this described somewhere in the docs though, without the necessity of investigating the debug log.

**Comment by pvalena** [comment]
Let me retest; I'll create some reproducer if this persists.

**Comment by psss** [comment]
The essential part of the use case should be covered by the `tmt try` command.
